src/qubx/backtester/simulator.py

```diff
@@ -187,13 +187,16 @@ def send_order(
         price: float | None = None,
         client_id: str | None = None,
         time_in_force: str = "gtc",
+        fill_at_price: bool = False,
```

Comment:
I'm thinking that this doesn't follow common interface contract for send_order if we add additional argument. May be it'd be clear if we just make interface extendable:
```
    def send_order(
        self,
        instrument: Instrument,
        order_side: str,
        order_type: str,
        amount: float,
        price: float | None = None,
        client_id: str | None = None,
        time_in_force: str = "gtc",
        **optional,                         # - all optional parameters here
    ) -> Order:
        raise NotImplementedError("send_order is not implemented")
```
and in SimulatedTrading : 
```
    def send_order(
        self,
        instrument: Instrument,
        order_side: str,
        order_type: str,
        amount: float,
        price: float | None = None,
        client_id: str | None = None,
        time_in_force: str = "gtc",
        **optional,
    ) -> Order:
        fill_at_price = optional.pop("fill_at_price", None)
        .....
```

Reply:
In this case we could handle additional parameters for live orders if need (for example routing settings for interactive brokers SMART route etc etc
)

Reply:
ah, I didn't see - you have already did that ))

Reply:
but I agree it is cleaner to pass **optional and unpack inside of the send_order